webapp/src/js/enketo/widgets/rdtoolkit-capture.js

```diff
@@ -0,0 +1,165 @@
+      return;
+    }
+
+    displayActions($widget);
+
+    $widget.on('click', '.btn.rdtoolkit-capture-test', () => captureRDTestResult($widget, rdToolkitService));
+  };
+
+  function captureRDTestResult($widget, rdToolkitService) {
+    const dateFormat = 'LLL';
+    const form = utils.getForm();
+    const sessionId = utils.getFieldValue(form, 'rdtoolkit-capture > rdtoolkit_session_id');
+
+    rdToolkitService
+      .captureRDTest(sessionId)
+      .then((response = {}) => {
+        const capturedTest = {
+          sessionId: response.sessionId || '',
+          state: response.state || '',
+          timeStarted: utils.formatDate(response.timeStarted, dateFormat),
+          timeResolved: utils.formatDate(response.timeResolved, dateFormat),
+          timeRead: utils.formatDate(response.timeRead, dateFormat),
+          results: response.results || [],
+          resultsDescription: getFormattedResult(response.results),
+          image: response.croppedImage
+        };
+
+        updateFields($widget, capturedTest);
+        hideActions($widget);
+        displayPreview($widget, capturedTest);
+      });
+  }
+
+  function displayActions($widget) {
+    window.CHTCore.Translate
+      .get('rdtoolkit.capture')
+      .toPromise()
+      .then(label => {
+        $widget
+          .find('.or-appearance-rdtoolkit-action-btn')
+          .after('<div class="rdtoolkit-preview"></div>')
+          .after(`
+            <div class="rdtoolkit-actions">
+              <a class="btn btn-primary rdtoolkit-capture-test">${label}</a>
+            </div>
+          `);
+      });
+  }
+
+  function hideActions($widget) {
+    $widget
+      .find('.rdtoolkit-actions')
+      .hide();
+  }
+
+  function displayPreview($widget, capturedTest) {
+    const imageTag = `<img src="data:image/png;base64,${capturedTest.image}">`;
+
+    $widget
+      .find('.rdtoolkit-preview')
+      .append(`
+        <div>
+          ${window.CHTCore.Translate.instant('report.rdtoolkit-capture.preview.title')}
+        </div>
+        <br>
+        <div>
+          <span class="rdt-label">
+            ${window.CHTCore.Translate.instant('report.rdtoolkit-capture.preview.results')} 
+          </span>
+          <span class="rdt-value">${capturedTest.resultsDescription}</span>
+        </div>
+        <div>
+          <span class="rdt-label">
+            ${window.CHTCore.Translate.instant('report.rdtoolkit-capture.preview.time_read')} 
+          </span>
+          <span class="rdt-value">${capturedTest.timeRead}</span>
+        </div>
+        <div>
+          <span class="rdt-label">
+            ${window.CHTCore.Translate.instant('report.rdtoolkit-capture.preview.image')} 
+          </span>
+          ${capturedTest.image ? imageTag : ''}
+        </div>
+        <br>
+        <div>
+           ${window.CHTCore.Translate.instant('report.rdtoolkit-capture.preview.next_action')} 
+        </div>
+      `);
+  }
+
+  function updateFields($widget, capturedTest) {
```

Comment:
I'm slightly concerned about hardcoding the model root, does this mean that we can't have two forms that use the new integration?

Reply:
Not sure if I'm getting it
Do you mean in the selectors to set and get values from the form? Can we display 2 forms at the same time? 😅

I tested provisioning 2 rdtests, then capturing the info in different order and closing without saving then opening a different one, etc. all with 1 form displayed at the time and it was ok. 
At some point I had 2 different forms type, using the same widget, displaying 1 form at the time when filling data and also worked.

Reply:
I mean different forms. For example, a project who would want to add an RD test in a malaria form, and their form name would be "malaria_rdt". And then later, they want a COVID RD test in a different form, called "covid_rdt". So they would need two separate forms for two different RD tests.

Reply:
If we have multiple and to make specific search without using a "namespace" (rdtoolkit-capture) then we can try adding an id (uuid) to the widget somehow but still not sure how to make that pretty <- not loving the idea yet

Reply:
Can you get the model root from the form somehow so you don't need to hardcode it?

Reply:
The form is already available to you, no?

Reply:
Oh I get it now, thanks, I'm going to test

Reply:
tagName of the model's rootElement 👍👍

```javascript
const form = utils.getForm();
const formTagName = form.model.rootElement.tagName;
const sessionId = utils.getFieldValue(form, `${formTagName} > rdtoolkit_session_id`);
.........
utils.setFieldValue($widget, `input[name="${formTagName}/data/rdtoolkit_time_read"]`, capturedTest.timeRead);
```

Reply:
Done, I've tested the same sample and it's working fine with 2 forms types using the same widget!